### PR TITLE
Only return an owner if the file exists + improved getUidAndFilename

### PR DESCRIPTION
**Pull request by icewind1991**
Move getUidAndFilename to the View so we don't have to implement it in 3 apps

cc @PVince81 @LukasReschke

### CODE REVIEW

File: lib/private/files/view.php [hunk 1]
```diff
@@ -1569,10 +1568,14 @@ private function searchCommon($method, $args) {
...
 	 *
 	 * @param string $path
-	 * @return string
+	 * @return string|null the user id of the owner or null if the file doesn't exist
 	 */
 	public function getOwner($path) {
-		return $this->basicOperation('getOwner', $path);
+		$info = $this->getFileInfo($path);
+		if (!$info) {
+			return null;
```
Comment (LukasReschke) [inline lib/private/files/view.php]: Any specific reason that we don't use an exception here? Might make bugs easier to catch?

Comment (icewind1991) [inline lib/private/files/view.php]: changed it to an exception

**Comment by LukasReschke** [comment]
@icewind1991 Tests are failing

**Comment by icewind1991** [comment]
all fixed

File: apps/files_versions/lib/storage.php [hunk 2]
```diff
@@ -41,6 +41,9 @@
 
 namespace OCA\Files_Versions;
 
+use OC\Files\Filesystem;
+use OC\Search\Provider\File;
+use OCA\Activity\Extension\Files;
```
Comment (LukasReschke) [inline apps/files_versions/lib/storage.php]: L45 and L46 seem unrequired?

Comment (icewind1991) [inline apps/files_versions/lib/storage.php]: removed

File: apps/files_trashbin/lib/trashbin.php [hunk 3]
```diff
@@ -62,7 +62,11 @@ class Trashbin {
 	 * @param array $params
 	 */
 	public static function ensureFileScannedHook($params) {
-		self::getUidAndFilename($params['path']);
+		try {
+			self::getUidAndFilename($params['path']);
+		} catch (NotFoundException $e) {
+
```
Comment (PVince81) [inline apps/files_trashbin/lib/trashbin.php]: ewww, log the error? Or at least add a comment why it is ok to ignore the error.

**Comment by PVince81** [comment]
Needs to be tested with encryption + sharing which intensively uses the owner info.

**Comment by PVince81** [comment]
Tested with encryption and the following cases: uploading into a shared folder as recipient, moving files into a shared folder as recipient and uploading to a public share. All work.

@icewind1991 please fix my comment above then this is good to go.

File: apps/files_trashbin/lib/trashbin.php (hunk 3 above)
Comment (icewind1991) [inline apps/files_trashbin/lib/trashbin.php]: > add a comment why it is ok to ignore the error

Done

**Comment by PVince81** [comment]
Thanks, :+1: from me.

Second reviewer @LukasReschke @schiesbn ?

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**